pkg/services/ngalert/store/alertmanager.go

```diff
@@ -78,7 +82,9 @@ func (st DBstore) SaveAlertmanagerConfigurationWithCallback(ctx context.Context,
 		if _, err := sess.Insert(config); err != nil {
 			return err
 		}
-
+		if _, err := st.deleteOldConfigurations(ctx, cmd.OrgID, ConfigRecordsLimit); err != nil {
```

Comment:
Do we need to run this on every save? We could run it a lot slower. Say every N saves, or perhaps every hour or several on a goroutine in the background? We could be even more aggressive and just run the cleanup once every restart of the server?